packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -64,8 +65,21 @@ export default class {
         servers => serversDb.update(servers)
       )
 
-      // Connects to existing servers.
       const servers = await serversDb.get()
+      const hosts = new Set(servers.map(_ => _.host))
+
+      // Add servers in XenStore
+      const xenStoreServers = await XenStore.read('vm-data/xen-servers')
+        .then(JSON.parse)
+        .catch(() => [])
+      for (const server of xenStoreServers) {
+        if (!hosts.has(server.host)) {
+          hosts.add(server.host)
+          servers.push(await this.registerXenServer(server))
+        }
+      }
```

Comment:
Question: should we always add unknown servers from the XenStore or only if there are no existing servers?

Reply:
I would say that if you already have servers, it means XOA is already started and then you could add new host "normally" (or via API).

So it would be logical to do it only when your XOA doesn't have any existing server, so this xenstore doesn't become a way to configure XOA post-first boot.

What do you think?

Reply:
I agree, that's why I asked.